types/queries.go

```diff
@@ -6,20 +6,29 @@ import (
 
 //-------- Queries --------
 
-type RawQueryResponse struct {
+// QueryResponse is the Go counterpart of `ContractResult<Binary>`.
```

Comment:
This is actually more complicated and fragile Go code. Using a wrapper type (as I did) is a typical case when partially overriding JSON marshalling, so you can dispatch to the default marshaller without the overrides in some conditions.

If you don't want to expose more types, I can rename `RawQueryResponse` to `rawQueryResponse` or `queryResponseHelper` so it is not seen outside the package. Just the QueryResponse type (wrapper)